modules/snooker/src/main/ts/ephox/snooker/api/Warehouse.ts

```diff
@@ -31,13 +32,34 @@ const filterItems = function (warehouse: Warehouse, predicate: (x: Structs.Detai
   return Arr.filter(all, predicate);
 };
 
+const generateColumns = <T extends Structs.Detail> (rowData: Structs.RowData<T>, columnsGroup: Record<string, Structs.Column>) => {
```

Comment:
You're mutating the `columnsGroup` record here and that goes against our functional programming principles. It doesn't look like this needs to be input and instead you could just return a new record.